packages/aft/test/version_bump/version_bump_test.dart

```diff
@@ -0,0 +1,228 @@
+        late Repo repo;
+        late String baseRef;
+
+        setUp(() async {
+          await copyDirectory(sourceDirectory, tempDirectory);
+          await GitDir.init(
+            tempDirectory.path,
+            allowContent: true,
+            initialBranch: 'main',
+          );
+          repo = await Repo.open(
+            AftConfigLoader(workingDirectory: tempDirectory).load(),
+          );
+          await runGit(repo, ['add', '.']);
+          await runGit(repo, ['commit', '-m', 'Initial commit\n']);
+          baseRef = await runGit(repo, ['rev-parse', 'HEAD']);
+        });
+
+        tearDown(() async {
+          if (generateSnapshots) {
+            final process = await repo.git.runCommand(['diff', '--unified=0']);
+            final diff = process.stdout as String;
+            await diffFile.writeAsString(processDiff(diff));
+          }
+          await tempDirectory.delete(recursive: true);
+        });
+
+        group('chore', () {
+          test('single amplify package', () async {
+            late String headRef;
+            for (final change in changes) {
+              headRef = await change.commit(repo);
+            }
+
+            await run([
+              'version-bump',
+              '--directory',
+              (tempDirectory.path),
+              '--base-ref',
+              (baseRef),
+              '--head-ref',
+              (headRef),
+              '--skip-build-version',
+            ]);
+
+            if (!generateSnapshots) {
+              final process = await repo.git.runCommand(
+                ['diff', '--unified=0'],
+              );
+              final actual = processDiff(process.stdout as String);
+              final expected = await diffFile.readAsString();
+              expect(actual, expected);
+            }
+          });
+        });
+      });
+    }
+  });
+}
+
+class Change {
+  const Change({
+    required this.title,
+    required this.packages,
+  });
+
+  final String title;
+  final List<String> packages;
+
+  Future<String> commit(Repo repo) async {
+    for (final package in packages) {
+      final newDir = Directory(repo[package].path).createTempSync();
+      File(p.join(newDir.path, 'file.txt')).createSync();
+    }
+    await runGit(repo, ['add', '.']);
+    await runGit(repo, [
+      'commit',
+      '-m',
+      '$title\n',
+    ]);
+    return runGit(repo, ['rev-parse', 'HEAD']);
+  }
+}
+
+Future<String> runGit(Repo repo, List<String> args) async {
+  final result = await repo.git.runCommand(args);
+  final stdout = result.stdout as String;
+  logger.info('git ${args.join(' ')}:\n$stdout');
+  return stdout.trim();
+}
+
+Future<void> copyDirectory(
+  Directory source,
+  Directory destination,
+) async {
+  for (final e in source.listSync(recursive: true)) {
+    if (!File(e.path).existsSync()) {
+      continue;
+    }
+    final path = e.path.replaceFirst(source.path, '');
+    logger.info('copying to ${destination.path}/$path');
```

Comment:
```suggestion
    logger.verbose('copying to ${destination.path}/$path');
```
nit: this is cluttering the test output and I don't see the value to always include.